### PR TITLE
detect-filesize: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: https://github.com/OISF/suricata/pull/8087
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4039) ticket:

Describe changes:

-Update detect-filesize unit tests to FAIL/PASS APIs

### CODE REVIEW

File: src/detect-filesize.c [hunk 1]
```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
...
+    FAIL_IF(snprintf(fullstr, 1024,
+                    "alert http any any -> any any (msg:\"Filesize "
+                    "test\"; filesize:%s; sid:1;)",
+                    str) >= 1024);
 
     *de_ctx = DetectEngineCtxInit();
-    if (*de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, fullstr);
```
Comment (victorjulien) [inline src/detect-filesize.c:262]: this line looks suspicious in 2 ways: `de_ctx` is used `*de_ctx` elsewhere in the test, but not here. 2nd: brackets mismatch :)

File: src/detect-filesize.c [hunk 2]
```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
...
 
     (*de_ctx)->flags |= DE_QUIET;
 
     (*de_ctx)->sig_list = SigInit(*de_ctx, fullstr);
```
Comment (victorjulien) [inline src/detect-filesize.c:266]: replace with `DetectEngineAppendSig` as well

File: src/detect-filesize.c [hunk 3]
```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
...
-    if (snprintf(fullstr, 1024, "alert http any any -> any any (msg:\"Filesize "
-                                "test\"; filesize:%s; sid:1;)", str) >= 1024) {
-        goto end;
-    }
+    FAIL_IF(snprintf(fullstr, 1024,
+                    "alert http any any -> any any (msg:\"Filesize "
+                    "test\"; filesize:%s; sid:1;)",
+                    str) >= 1024);
 
     *de_ctx = DetectEngineCtxInit();
```
Comment (victorjulien) [inline src/detect-filesize.c:261]: needs a check as well

File: src/detect-filesize.c [hunk 4]
```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
...
-    if (*de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, fullstr);
 
     (*de_ctx)->flags |= DE_QUIET;
```
Comment (victorjulien) [inline src/detect-filesize.c:264]: this should directly follow the `DetectEngineCtxInit`+its check

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by hsadia538** [comment]
Followed by #8104

**Pull request closed without merging**